src/solver/application/application.cpp

```diff
@@ -176,21 +163,48 @@ void Application::prepare(int argc, char* argv[])
 
     checkFuelCostColumnNumber(pStudy->areas);
     checkCO2CostColumnNumber(pStudy->areas);
+}
 
-    // Start the progress meter
-    pStudy->initializeProgressMeter(pSettings.tsGeneratorsOnly);
-    if (pSettings.noOutput)
-        pSettings.displayProgression = false;
+void Application::prepare(int argc, char* argv[])
+{
+    pArgc = argc;
+    pArgv = argv;
 
-    if (pSettings.displayProgression)
-    {
-        auto& filename = pStudy->buffer;
-        filename.clear() << "about-the-study" << Yuni::IO::Separator << "map";
-        pStudy->progression.saveToFile(filename, *resultWriter);
-        pStudy->progression.start();
-    }
-    else
-        logs.info() << "  The progression is disabled";
+    // Load the local policy settings
+    LocalPolicy::Open();
+    LocalPolicy::CheckRootPrefix(argv[0]);
+
+    Resources::Initialize(argc, argv);
```

Comment:
seems that this method cares only about argv 0 it will be clearer if you give it a const char* or const std::string or something like that?

Reply:
Not really
```cpp
if (argc < 1 or argv[0] == NULL)
{
    logs.error() << "Impossible to find the root folder";
    return;
}
```

Reply:
is this the method responsability?

Reply:
This a larger topic about CLI options management (and ini parameters)